### PR TITLE
Add statefulset upgrade tests to cluster_upgrade

### PR DESCRIPTION
**Pull request by bdbauer**
**What this PR does / why we need it**:
Adds already created statefulset upgrade tests to cluster_upgrade.go. With further test infra changes, this will allow them to be continuously run, giving better signals.

Detect and prevent issues like https://github.com/kubernetes/kubernetes/issues/48327

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/lifecycle/cluster_upgrade.go [hunk 1]
```diff
@@ -177,7 +183,35 @@ var _ = SIGDescribe("etcd Upgrade [Feature:EtcdUpgrade]", func() {
 				defer finalizeUpgradeTest(start, etcdTest)
 				framework.ExpectNoError(framework.EtcdUpgrade(framework.TestContext.EtcdUpgradeStorage, framework.TestContext.EtcdUpgradeVersion))
 			}
-			runUpgradeSuite(f, testFrameworks, testSuite, upgCtx, upgrades.EtcdUpgrade, upgradeFunc)
+			runUpgradeSuite(f, upgradeTests, testFrameworks, testSuite, upgCtx, upgrades.EtcdUpgrade, upgradeFunc)
+		})
+	})
+})
+
+var _ = SIGDescribe("stateful Upgrade [Feature:StatefulUpgrade]", func() {
```
Comment (krousey) [inline test/e2e/lifecycle/cluster_upgrade.go]: This is going to put "[sig-cluster-lifecycle]" in the name when I don't think we're going to own it. Can you use the standard describe and add an owning sig here?

File: test/e2e/lifecycle/cluster_upgrade.go [hunk 2]
```diff
@@ -177,7 +183,35 @@ var _ = SIGDescribe("etcd Upgrade [Feature:EtcdUpgrade]", func() {
...
+	// Create the frameworks here because we can only create them
+	// in a "Describe".
+	testFrameworks := createUpgradeFrameworks(statefulsetUpgradeTests)
+	framework.KubeDescribe("stateful upgrade", func() {
+		It("should maintain a functioning cluster", func() {
+			upgCtx, err := getUpgradeContext(f.ClientSet.Discovery(), framework.TestContext.UpgradeTarget)
+			framework.ExpectNoError(err)
+
+			testSuite := &junit.TestSuite{Name: "Stateful upgrade"}
+			statefulUpgradeTest := &junit.TestCase{Name: "stateful-upgrade", Classname: "upgrade_tests"}
```
Comment (krousey) [inline test/e2e/lifecycle/cluster_upgrade.go]: Name needs a sig tag.

**Review by krousey — CHANGES_REQUESTED** [review]
The Name() method of all the upgrade tests should also have a sig tag to indicate ownership.

**Comment by foxish** [comment]
Thanks!

/lgtm
/approve

A test of this setup being correct and working would be to detect the upgrade issue between 1.6 and 1.7 tracked in https://github.com/kubernetes/kubernetes/issues/48327.

**Pull request merged**